gh-141004: Document `PyFunction_SetKwDefaults` (#141294)
python/cpython: Doc/c-api/function.rst

   dictionary of arguments or ``NULL``.

> .. c:function:: PyObject* PyFunction_SetKwDefaults(PyObject *op, PyObject *defaults)

Reviewer: ```suggestion
.. c:function:: int PyFunction_SetKwDefaults(PyObject *op, PyObject *defaults)
```
You must have been looking at `PyFunction_GetKwDefaults`

Author: Ah, you're right, let me fix this locally.

Author: Addressed, sorry for not using the suggestions, but I had different wording in mind.

Reviewer: > sorry for not using the suggestions, but I had different wording in mind.

That's fine, I agree it's better.

   Set the keyword-only argument default values of the function object *op*.

>    *defaults* must be a dictionary of keyword-only arguments or ``Py_None``.

Reviewer: ```suggestion
   *defaults* must be a dictionary of keyword-only arguments or :c:data`Py_None`.
```

Author: We didn't link to this above in `PyFunction_SetDefaults`, so I don't really want to be inconsistent here. It's usually not that important to add refs for `Py_None` anyway.

> 

Reviewer: ```suggestion
   Returns ``0`` on success, or ``-1`` on failure with an exception set.

```

Author: I'm not a huge fan of the wording here, so I addressed this using my own.

   dictionary of arguments or ``NULL``.

.. c:function:: int PyFunction_SetKwDefaults(PyObject *op, PyObject *defaults)

>    Set the keyword-only argument default values of the function object *op*.
>    *defaults* must be a dictionary of keyword-only arguments or ``Py_None``.

Reviewer: ```suggestion
   Set the keyword-only parameter default values of the function object *op*.
   *defaults* must be a dictionary of keyword-only parameters or ``Py_None``.
```

Author: We say "argument" everywhere else in this file, so I'd rather not deviate from it here. I don't think that the distinction is particularly helpful for readers here.

Reviewer: I'm not a fan of such a precedent, but unfortunately it is. Fixing this is tracked in https://github.com/python/cpython/issues/133438.
